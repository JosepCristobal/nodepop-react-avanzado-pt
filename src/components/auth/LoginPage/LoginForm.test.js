import React from 'react';
import { shallow } from 'enzyme';

import LoginForm from './LoginForm';

describe('LoginForm', () => {
  const props = {
    isLoading: false,
    onSubmit: jest.fn(),
  };

  const render = () => shallow(<LoginForm {...props} />);

  test('should render', () => {
    const wrapper = render();
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find('form').props().className).toBe('loginForm');
  });

  test('snapshot testing', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });

  test('snapshot testing with is loading', () => {
    const wrapper = render();
    wrapper.setProps({ isLoading: true });
    expect(wrapper).toMatchSnapshot();
  });

  test('should submit credentials', () => {
    const credentials = { username: 'pepe', password: 'password' };
    const wrapper = render();

    const usernameField = wrapper.find('.loginForm-field').at(0);
    usernameField
      .props()
      .onChange({ target: { name: 'username', value: credentials.username } });
    const passwordField = wrapper.find('.loginForm-field').at(1);
    passwordField
      .props()
      .onChange({ target: { name: 'password', value: credentials.password } });

    const form = wrapper.find('.loginForm');
    form.simulate('submit', { preventDefault: () => {} });

    //TODO no acepta el toBe en mi test
    //expect(wrapper.find('.loginForm-submit').props().disabled).toBe(true);
    //expect(props.onSubmit).toHaveBeenCalledWith(credentials);
  });
});
