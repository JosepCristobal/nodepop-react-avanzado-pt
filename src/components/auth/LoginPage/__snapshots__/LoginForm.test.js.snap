// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot testing 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <FormField
    autofocus={true}
    className="loginForm-field"
    label="Username"
    name="email"
    onChange={[Function]}
    type="text"
    value=""
  />
  <FormField
    className="loginForm-field"
    label="password"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <CheckField
    checked={false}
    className="Checkbox"
    label="Save session "
    name="check"
    onChange={[Function]}
    type="checkbox"
  />
  <styled.button
    className="loginForm-submit"
    disabled={true}
    type="submit"
    variant="primary"
  >
    Log in
  </styled.button>
</form>
`;

exports[`LoginForm snapshot testing with is loading 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <FormField
    autofocus={true}
    className="loginForm-field"
    label="Username"
    name="email"
    onChange={[Function]}
    type="text"
    value=""
  />
  <FormField
    className="loginForm-field"
    label="password"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <CheckField
    checked={false}
    className="Checkbox"
    label="Save session "
    name="check"
    onChange={[Function]}
    type="checkbox"
  />
  <styled.button
    className="loginForm-submit"
    disabled={true}
    type="submit"
    variant="primary"
  >
    Log in
  </styled.button>
</form>
`;
